File: basic data structure code in c/Basic data structure/stack using array.c
#include<stdio.h>

void main() {
  int size, choice;
  printf("insert the size of array : ");
  scanf("%d", & size);
  int top = -1;
  int arr[size];

  while (choice != -1) {
    printf("\n*************************************\n");
    printf("press 1 for push data in stack \n");
    printf("press 2 for pop data in stack \n");
    printf("press 3 for peek data in stack\n ");
    printf("press 4 to  display the whole stack \n ");
    printf("press -1 to exit \n");
    printf("Enter your choice : ");
    scanf("%d", & choice);

    if (choice == 1) {
      if (top == size - 1) {
        printf("\t\t\tWarning:The stack is full");
        break;
      } else {
        push(arr, size, top);
      }
      top = top + 1;
    }

    if (choice == 2) {
      if (top == -1) {
        printf("\t\t\tWarning:The stack is empty");
        break;
      } else {
        pop(arr, size, top);
      }
      top = top - 1;
    }
    if (choice == 3) {
      printf("the top most element of stack is : %d", arr[top]);
    }
    if (choice == 4) {
      display(arr, top);
    }
  }
}

int push(int arr[], int size, int top) {
  int item, i;
  printf("\n insert data item : ");
  scanf("%d", & item);
  top = top + 1;
  arr[top] = item;

  printf("\n\n");
  printf("after data insrted into stack : \n");
  for (i = 0; i < top; i++) {
    printf("%d     ", arr[i]);
  }

}

int pop(int arr[], int top, int size) {
  int i, temp;
  temp = arr[top];
  printf("\n\n after data deleted from stack : \n");
  for (i = 0; i < size; i++) {
    printf("%d    ", arr[i]);
  }
}

int display(int arr[], int top) {
  int i;
  for (i = 0; i <= top; i++) {
    printf("%d   ", arr[i]);

  }
}

File: basic data structure using python/stack_using_array.py
def push(item):
    a[top]=item

def pop ():
    a[top]=0

def display():
    print("your data is  : ")
    for i in range(len(a)):
        print(a[i], end="  ")
    print(" ")

def isempty():
    return top>0

def isfull():
    return top<len(a)

if __name__ == '__main__':

    a = [0] * 6
    top = int(0)
    choice = int()
    while (choice != -1):
        print("1. INSERT DATA ")
        print("2. DELETE DATA")
        print("3. DISPLAY DATA")

        choice = int(input("\t\t\t PLEASE ENTER YOUR CHOICE : "))
        if(choice==1):
            if(isfull()):
                item=int(input("enter the data element  :  "))
                push(item)
                top=top+1
            else:
                print("\t\t\t\t\t\t\t\t\t\t\t\t STACK IS FULL\n")
        if(choice==2):
            if(isempty()):
                top = top - 1
                pop()
            else:
                print(" \t\t\t\t\t\t\t\t\t\t\t\tSTACK IS EMPTY\n")

        if(choice==3):
            display()
